core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1544,12 +1544,14 @@ class ReplicaManager(val config: KafkaConfig,
                                  fetchPartitionStatus: Seq[(TopicIdPartition, FetchPartitionStatus)]): Option[LogReadResult] = {
     val key = new TopicPartitionOperationKey(remoteFetchInfo.topicPartition.topic(), remoteFetchInfo.topicPartition.partition())
     val remoteFetchResult = new CompletableFuture[RemoteLogReadResult]
-    var remoteFetchTask: Future[Void] = null
+    var remoteFetchFuture: RemoteFetchFuture = null
     try {
-      remoteFetchTask = remoteLogManager.get.asyncRead(remoteFetchInfo, (result: RemoteLogReadResult) => {
+      val remoteLogReader = remoteLogManager.get.remoteLogReaderTask(remoteFetchInfo, (result: RemoteLogReadResult) => {
         remoteFetchResult.complete(result)
         delayedRemoteFetchPurgatory.checkAndComplete(key)
       })
+      val remoteFetchTask = remoteLogManager.get.asyncRead(remoteLogReader)
+      remoteFetchFuture = new RemoteFetchFuture(remoteLogReader, remoteFetchTask)
```

Comment:
Hmm, is this correct? We need to pass in remoteFetchFuture to RemoteFetchFuture, right?

Reply:
I think you mean the result future right? If yes; agree, I missed it.
I'm passing both future as input now.